libibverbs/device.c

```diff
@@ -262,6 +267,8 @@ struct ibv_context *__ibv_open_device(struct ibv_device *device)
 	context->cmd_fd = cmd_fd;
 	pthread_mutex_init(&context->mutex, NULL);
 
+	ibverbs_device_hold(device);
+
```

Comment:
We made some extra checking re the option to use the openat system call and found that it can be done. However, it requires holding  an open 'FD' per ibv_device.  In some clouds environments which uses SRIOV and many NIC(s) it might be large numbers of 'FD's which is a limited resource per process.  The alternative solution with TS doesn't require to hold any additional 'FD' and as such is preferred,